PyPI_analysis/detect_imports.py

```diff
@@ -214,75 +200,101 @@ def dynamic_imports(parsed_code: ast.Module):
+                    for node_import in node_ds.body:
+                        if isinstance(node_import, ast.Import):
+                            for alias in node_import.names:
+                                name = alias.name.split(".", 1)[0]
+                                if is_external_import(name):
+                                    yield {
+                                        "Docstring": ParsedImport(
+                                            name=name,
+                                            source=source.supply(
+                                                lineno=node.body[0].lineno
+                                            ),
+                                        )
+                                    }
+                        elif isinstance(node_import, ast.ImportFrom):
+                            # Relative imports are always relative to the current package, and
+                            # will therefore not resolve to a third-party package.
+                            # They are therefore uninteresting to us.
+                            if (
+                                node_import.level == 0
+                                and node_import.module is not None
+                            ):
+                                name = node_import.module.split(".", 1)[0]
+                                if is_external_import(name):
+                                    yield {
+                                        "Docstring": ParsedImport(
+                                            name=name,
+                                            source=source.supply(
+                                                lineno=node.body[0].lineno
+                                            ),
+                                        )
+                                    }
+
+    def regular_imports(node: ast.AST):
+        if isinstance(node, ast.Import):
+            logger.debug(ast.dump(node))
+            for alias in node.names:
+                name = alias.name.split(".", 1)[0]
+                if is_external_import(name):
+                    yield {
+                        "Regular": ParsedImport(
+                            name=name, source=source.supply(lineno=node.lineno)
+                        )
+                    }
+        elif isinstance(node, ast.ImportFrom):
+            logger.debug(ast.dump(node))
+            # Relative imports are always relative to the current package, and
+            # will therefore not resolve to a third-party package.
+            # They are therefore uninteresting to us.
+            if node.level == 0 and node.module is not None:
+                name = node.module.split(".", 1)[0]
+                if is_external_import(name):
+                    yield {
+                        "Regular": ParsedImport(
+                            name=name, source=source.supply(lineno=node.lineno)
+                        )
+                    }
 
```

Comment:
I assume this is more of less a copy of the corresponding code in FD's `extract_imports.py`?

Reply:
exactly